### PR TITLE
Add seed data and generate in development

### PR DESCRIPTION
**Pull request by asungur**
- Don't auth `/data/*` requests in development (up to you, I think its useful for testing)
- Improve `/collections/*` controller actions
- Add seed data

### CODE REVIEW

File: src/db/index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const mongoose = require('mongoose');
 const errorHandler = require('../utils/errorHandler');
 const mongoOperator = require('../models/mongoOperator');
 
-const DEFAULT_COLLECTION_NAMES = ['cloudCodeFunctions', 'files', 'users'];
+const DEFAULT_COLLECTION_NAMES = ['cloudcodefunctions', 'files', 'users'];
```
Comment (asungur) [inline src/db/index.js:5]: This was showing up as an additional collection

File: src/models/User.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ const userSchema = new mongoose.Schema({
     type: String,
     unique: true,
     trim: true,
-    minLength: 8,
```
Comment (asungur) [inline src/models/User.js:14]: User names like Jo, Bo should be allowed as long as they are unique

File: src/utils/userAuthentication.js [hunk 3]
```diff
@@ -46,7 +46,7 @@ const hashPassword = async (plaintextPassword) => {
 };
 
 const checkAuthenticated = (req, res, next) => {
-  if (req.isAuthenticated()) {
+  if (req.isAuthenticated() || config.NODE_ENV !== 'production') {
```
Comment (asungur) [inline src/utils/userAuthentication.js:49]: I thought this would be helpful for testing stuff. Let me know what you guys think

Comment (Reillyknut) [reply src/utils/userAuthentication.js:49]: Yeah I'm a big fan of this

**Review by Reillyknut — APPROVED** [review]

**Pull request merged**